src/ngraph/ops/slice.cc

```diff
@@ -0,0 +1,92 @@
+/*******************************************************************************
+* Copyright 2018 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "ops/slice.h"
+#include "../operator/tensor/matrix_op-inl.h"
+#include "ngraph_sgcompiler_utils.h"
+
+namespace ngraph_bridge {
+
+NgraphNodePtr create_slice_op(const NgraphNodePtr& node,
+                              const nnvm::NodeAttrs& attrs) {
+  const mxnet::op::SliceParam& param =
+      nnvm::get<mxnet::op::SliceParam>(attrs.parsed);
+  nnvm::TShape tshape = NShape_to_TShape(node->get_shape());
+  ngraph::Coordinate ng_begin, ng_end, ng_step;
+  ngraph::AxisSet axes;
+  bool reverse = false;
+
+  for (mxnet::index_t i = 0; i < param.begin.ndim(); ++i) {
+    const int len = tshape[i];
+
+    int s = 1;
+    if (param.step[i].has_value()) {
+      s = param.step[i].value();
+      if (s == 0) s = 1;
```

Comment:
Style suggestion: Always use `{` and `}`, even with single-statement bodies.  IME it's an overall productivity win, when considering the bug-hunting it saves.

This comment applies to a few additional lines of code further down.